### ISSUE TITLE
Playwright >> Some violations not reported by sdk when no rules are specified.

### ISSUE THREAD

**Issue by quldude** [dequelabs/axe-core-npm#830]
### Product

playwright

### Product Version

4.7.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
Reports following violations:
![expected](https://github.com/dequelabs/axe-core-npm/assets/21183880/aa3e9d7d-5d05-4c38-b4e2-f31b1e763353)

#### Actual
The sdk does not report violations for `IDs of active elements must be unique` and `id attribute value must be unique`.

#### How to Reproduce
The webpage under test shows the following violations when using chrome extension `axe DevTools`:
![expected](https://github.com/dequelabs/axe-core-npm/assets/21183880/aa3e9d7d-5d05-4c38-b4e2-f31b1e763353)

But the playwright test does not detect 2 of the violations as mentioned in Actual.
Playwright test code:
```
const accessibilityScanResults = await new AxeBuilder({ page }).analyze();

await testInfo.attach("get-started-accessibility-scan-results", {
  body: JSON.stringify(accessibilityScanResults, null, 2),
  contentType: "application/json",
});

expect(
  accessibilityScanResults.violations,
  "No violations present in Get Started page.",
).toEqual([]);
```

#### Additional context
If I use the code below then the 2 missing violations are reported:
```
new AxeBuilder({ page }).withRules([
        "duplicate-id",
        "duplicate-id-active",
      ]);
```

**Comment by Zidious** [comment]
Hey @quldude, 

Is the page you're testing publicly available? If you run your test in headed mode does do you have any popups or modals that require the user interaction that might affect what is further loaded onto the page when `analyze()` is called?

**Comment by quldude** [comment]
I can't give the test site I'm using but you can see a similar issue with this page: https://ppsapplication.permitium.com/ppsb/start

The chrome extensions results in this:
![image](https://github.com/dequelabs/axe-core-npm/assets/21183880/b6f10f2e-432f-4a6a-a771-64e06c9e2b05)

But the sdk is missing the following violation:
`IDs of active elements must be unique`

**Comment by straker** [comment]
So the issue looks like that @axe-core/playwright is currently using axe-core 4.8, which [disabled duplicate-id-active](https://github.com/dequelabs/axe-core/releases/tag/v4.8.0). The extension has not yet made the update so is still running axe-core 4.7 where the rules are active. The extension should release soon at which point it will be using axe-core 4.8 as well.

If consistency between the two is important, you can install axe-core 4.7 manually to run with playwright and then it will then show duplciate-id-active violations, but you'll need to undo the install later when the extension updates.

**Comment by quldude** [comment]
I liked having duplicate-id / duplicate-id-active rules so developers know not to do that. I'm not understanding why they were disabled?

**Comment by quldude** [comment]
I found why there were disabled: https://www.deque.com/blog/wcag-2-2-removes-4-1-1-parsing-and-how-axe-core-is-impacted/. Will wait for the extension to be updated.

**Comment by straker** [comment]
@scurker ☝️

**Comment by straker** [comment]
@quldude  `section508` rules should be available. Any [rule tagged section508](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md) should be available to run.

**Comment by schne324** [comment]
@quldude, selecting section 508 is not currently supported in the extension. It is something we are considering adding support for but I do not currently have a timeline on this feature.

**Comment by padmavemulapati** [comment]
Here validated the scenarios:
1. With Production version of axe-core-npm packages(4.7.0), holding a axe-core version 4.7.2 (old version) with `^` in dependencies.

![image](https://github.com/dequelabs/axe-core-npm/assets/47409579/79f8cf91-84e7-44ee-b366-3bf250a4124e)

in the output it is getting the axe-core version with latest axe-core(4.8.2) because of the dependency specified as `^` so it allowing the next to 4.7.2 all minor versions.

![image](https://github.com/dequelabs/axe-core-npm/assets/47409579/450a9e03-3ef6-48af-9b33-b0e4ba4ddfdc)

2. With Production version of axe-core-npm pacakges(4.7.0), holding a axe-core version 4.7.2(old version) without `^` in the dependencies.
<img width="452" alt="image" src="https://github.com/dequelabs/axe-core-npm/assets/47409579/a56790a9-ed1c-4c18-874f-e1e5bdb11224">

in the output it is getting the axe-core version with  (4.7.2) because of the dependency specified without `^`  it is strictly considering the specified version and not allowing the next to 4.7.2 all minor versions.

![image](https://github.com/dequelabs/axe-core-npm/assets/47409579/7f3df09d-880b-4484-8eba-781aba68bcde)

3. With the latest canary version of axe-core-npm packages(4.7.4-f03670b.0), and  with  (4.7.2) because of the dependency specified without `^`  it is strictly considering the specified version and not allowing the next to 4.7.2 all minor versions.

<img width="459" alt="image" src="https://github.com/dequelabs/axe-core-npm/assets/47409579/a176b4bf-33a5-4231-945d-c0635b1129d9">

in the output it is getting the axe-core version with  (4.7.2) because of the dependency specified without `^`  it is strictly considering the specified version and not allowing the next to 4.7.2 all minor versions.

![image](https://github.com/dequelabs/axe-core-npm/assets/47409579/72db0172-0443-4ccb-9290-c69cdaf0f793)

4.  With the latest canary version of axe-core-npm packages(4.7.4-f03670b.0) , and  with latest (4.8.2) because of the dependency specified without `^`  it is strictly considering the specified version and not allowing the next to 4.8.2 all minor versions.

<img width="463" alt="image" src="https://github.com/dequelabs/axe-core-npm/assets/47409579/abe15855-77a4-41c1-9f42-fa80335472dd">

in the output it is getting the axe-core version with  (4.8.2) because of the dependency specified without `^`  it is strictly considering the specified version and not allowing the next to 4.8.2 all minor versions.

![image](https://github.com/dequelabs/axe-core-npm/assets/47409579/53ce24af-b18a-446f-b4b6-303e7f25ae52)


## Environment:
| **Label**                         | **Value** |
| -------------------------- | ------ |
| **Product**                | _axe-core-npm_|
|**Version** | _4.7.4-f03670b.0_ |
| **OS-Details**      | _MAC- Intel Core i7 - 11.6.8  & Windows11 _ |
|**BrowserDetails**|_Chrome Version 117.0.5938.88 (Official Build) (64-bit) & Firefox 117.0_|
